Week-4/OOP_js/Practice_ES6/script.js

```diff
@@ -0,0 +1,122 @@
+//Session-1 Inheritance
+class Vehicle {
+  constructor(Make, Model, Year, Color) {
+    this.make = Make;
+    this.model = Model;
+    this.year = Year;
+    this.color = Color;
+  }
+  drive() {
+    console.log(`Driving {${this.make}} {${this.model}}`);
+  }
+}
+
+class Car extends Vehicle {
+  constructor(Make, Model, Year, Color, NumSeats) {
+    super(Make, Model, Year, Color);
+    this.numSeats = NumSeats;
+  }
+}
+
+class RideShareCar extends Car {
+  constructor(Make, Model, Year, Color, NumSeats, isAvail) {
+    super(Make, Model, Year, Color, NumSeats);
+    this.isAvaliable = isAvail;
+  }
```

Comment:
same comment as prev

Reply:
please keep name convention consistent, otherwise it can be a mix of camelCase and CapitalCamelCase which isn't a good practice.

Reply:
Hmm, i probably didn't want the property and argument to have the same characters, so i will change the argument to something else i guess?

Reply:
you can have them same, and using this keyword will set it to current instance. It won't lead to any conflict.